### ISSUE TITLE
Question: Connecting to SAP Hana DB using JdbcSlim

### ISSUE THREAD

**Issue by ybermejo** [six42/jdbcslim#28]
Hi,

I'm trying to connect FitNesse with SAP Hana database using jdbslim. I followed the steps here https://help.sap.com/viewer/0eec0d68141541d1b07893a39944924e/2.0.00/en-US/ff15928cf5594d78b841fbbe649f04b4.html

I placed the ngdbc.jar in fitnesse/fixtures. Here's my code but when testing it, I'm having error. Really need help on this. Thanks in advance! I just scrubbed some details that's why there's a blank sections.
![image](https://user-images.githubusercontent.com/40356873/43083791-dfcaa748-8ec9-11e8-80d3-3d9c95cc0c8c.png)

**Comment by Unforgettable631** [comment]
Is your db url correct? It looks like it starts with ‘dbc’ instead of ‘jdbc’?

What kind of error shows up if you click on ‘Execution Log’ (green button at the top of the page)? 

You can also get more info adding |debug|true| in your defined properties.

It should be possible to connect to sap Hana with jdbcSlim, we did put the driver from the supplier (coming with the installation of SAP Hana) on our classpath.

**Comment by ybermejo** [comment]
Hi @Unforgettable631 ,

Yeah I missed that and changed to jdbc but still having an error. Says that,
![image](https://user-images.githubusercontent.com/40356873/43087167-c0dcb1e8-8ed1-11e8-87f0-9f053f853ae5.png)

May i ask where did you put the driver? Thanks.

**Comment by Unforgettable631** [comment]
Sure, we’ve put all local jars to a lib folder and added that directory
into our classpath. All dependency jars (coming via Maven central) are
placed in the fixture folder defined in a pom file.

Can you make Sure you have the correct credentials to access the db url you
provided? It could be you only have access to a specific scheme and
therefore getting a connection refused.

Which hana version do you want to test and are you using tenants? I can
check our configuration in detail when I’m back from holiday.


Op ma 23 jul. 2018 om 23:41 schreef ybermejo <notifications@github.com>

**Comment by ybermejo** [comment]
Hi @Unforgettable631 ,

Can you elaborate this? Sorry I'am new to this.
> Sure, we’ve put all local jars to a lib folder and added that directory
into our classpath. All dependency jars (coming via Maven central) are
placed in the fixture folder defined in a pom file.

I'm using correct credentials.
Version: 1.00.122.04.1478575636
Using tenant

**Comment by Unforgettable631** [comment]
Sure, but as I mentioned that Will take Some while. But I think it has something to do with your portnumber (also based on the error you get). 

Are you using the right instance number (by default the hana cliënt uses 30015, Then the system db should listen to that port as well. For tenants it’s 30013. But connection from an other tool such as dbvisualizer or in this case FitNesse, an instance number is probably missing.) The instance number is placed at the zero’s, f.e. If you have portnumber 31013 then 10 is your instancenumber.
 
Can you check if the portnumber is correct? You are getting sql exception -813 right?

**Comment by ybermejo** [comment]
Hi @Unforgettable631,
Can you confirm if I'm using the correct port based on the attached images?
![image](https://user-images.githubusercontent.com/40356873/43150042-86e487f2-8f9b-11e8-9c50-b1cc0c382db5.png)

**Comment by ybermejo** [comment]
Hi @Unforgettable631 ,

I tried to change the port number with 30013 and I got this sql exception, what does it means?
![image](https://user-images.githubusercontent.com/40356873/43190134-e559f6ac-902a-11e8-927f-8d820455ce2d.png)

**Comment by Unforgettable631** [comment]
I can’t load the images unfortunately due to bad WiFi connection. Can you try connecting with the Same configuration via dbvisualizer or so? Please let me know if you can connect via an other tool Then sap hana cliënt. 

Just in case, you are using a user/ password right?

**Comment by ybermejo** [comment]
Just login in SAP Hana Studio with the same config/credentials and it passes through.

**Comment by Unforgettable631** [comment]
Yes, sap hana cliënt will automatically redirect you to the configured in-/outbound Ports for your usage.

Please try to connect with an other tool instead of the sap hana cliënt (such as dbvisualizer or so). Then we can confirm the settings are correct (see also https://www.sap.com/developer/tutorials/hxe-connect-hxe-using-jdbc.html specific the information about port Numbers with MDC).

**Comment by Unforgettable631** [comment]
Glad you got it working now. I know setup can take a bit of time, especially SAP HANA db.
